src/com/mellanox/cicd/Matrix.groovy

```diff
@@ -225,34 +225,56 @@ def gen_image_map(config) {
         config.runs_on_dockers.each { item ->
 
             def dfile = item.clone()
-            dfile.arch = dfile.arch ?: arch
+            config.logger.debug("run on dockers item: " + dfile)
+            config.logger.debug("run on dockers item.name: " + dfile.name)
+            config.logger.debug("run on dockers item.enable: " + dfile.enable)
+
+            if (dfile.enable == null) {
+                dfile.enable = "true"
+                config.logger.debug("run on dockers item.enable: " + dfile.enable)
+            }
 
-            if (dfile.arch && dfile.arch != arch) {
-                config.logger.trace(3, "skipped conf: " + arch + " name: " + dfile.name)
-                return
+            if (dfile.enable == "auto") {
```

Comment:
why needed? what is a use-case you are after? this logic is pars of loop for each item in ```runs on dockers```, if some item is undesired - can just comment it?
please describe use-case/problem, what logic is expected for ```auto```?

Reply:
Developers want per-OS check boxes in job parameters without requirement to commit it (when you comment the line, you have to commit it). For example, they often want to build specific release package with OS1, OS2 (both are enabled by default) and OS3 (disabled by default) by using just job params check boxes.

Reply:
i see, makes sense.

Reply:
We have a jenkins job for build several packages for a lot of OS and architectures. If we need to build a special package we use step enable/disable mechanism but if we need to rebuild all or one package for only one OS we need to rebuild ALL. This is problem. 

We add new field "enable" to "run_on_dockers" record. If the value true that we run on this docker. This value can be an environment variable. For record with name "ubuntu-20.04-5.5-x86_64" the good idea set variable with name "ubuntu-20.04-5.5-x86_64" but this is double of code. So "auto" is syntax sugar. If value is "auto" then it will be replace to name of record(run_on_dockers) automatically and checked this environment variable.